docs/proposals/002-api-proposal/proposal.md

```diff
@@ -0,0 +1,339 @@
+
+### BackendPool
+
+The BackendPool at its core is a logical grouping of compute, expressed in the form of Pods (typically model servers), akin to a K8s Service. The BackendPool would deploy its own routing, and offer administrative configuration to the Platform Admin. 
+
+ It is expected for the BackendPool to:
+ - Enforce fair consumption of resources across competing use cases
+ - Efficiently route requests across shared compute (as displayed by the PoC)
+ 
+It is _not_ expected for the BackendPool to:
+ - Enforce any common set of adapters or base models are available on the Pods
+ - Manage Deployments of Pods within the Pool
+ - Manage Pod lifecycle of pods within the pool 
+
+Additionally, any Pod that seeks to join a BackendPool would need to support a protocol, defined by LLM Instance Gateway, to ensure the Pool has adequate information to intelligently route requests.
+
+### ModelGroup
+
+A ModelGroup allows the UseCaseOwner to define:
+- Which LoRA adapter(s) to consume 
+  - ModelGroup allows for traffic splitting between adapters _in the same BackendPool_ to allow for new LoRA adapter versions to be easily rolled out 
+- SLO objectives for the UseCase
+- The Pools this UseCase is relevant to 
+
+### Spec
+
+**ModelGroup**
+```golang
+// ModelGroup represents a set of LLM use cases that are multiplexed onto one 
+// or more backend pools. This resource is managed by the "Model Use Case Owner"
+// persona. The model use case owner a team that trains, verifies, and
+// leverages a large language model from a model frontend, drives the lifecycle
+// and rollout of new versions of those models, and defines the specific
+// performance and latency goals for the model. These model use cases are
+// expected to operate within a BackendPool sharing compute capacity with other
+// ModelUseCases, defined by the Inference Platform Admin. We allow a user who
+// has multiple UseCases across multiple pools (with the same config) to
+// specify the configuration exactly once, and deploy to many pools 
+// simultaneously. Enabling a simpler config and single source of truth
+// for a given user. ModelUseCase names are unique for a given BackendPool,
+// if the name is reused, an error will be  shown on the status of a 
+// ModelGroup that attempted to reuse. The oldest ModelUseCase, based on
+// creation timestamp, will be selected to remain valid. In the event of a race
+// condition, one will be selected at random. 
+type ModelGroup struct {
+        metav1.ObjectMeta
+        metav1.TypeMeta
+
+        Spec ModelGroupSpec
+}
+
+type ModelGroupSpec struct {
+        // Defines the use cases in the set.
+        // UseCases can be in 2 priority classes, Critical and Noncritical. 
+        // Priority class is implicitly set to Critical by specifying an Objective.
+        // Otherwise the UseCase is considered Noncritical.
+        UseCases   []ModelUseCases
```

Comment:
I think we should rename this to `models`.  Each of these is about a public model (so the field should be `name`, not `modelName`).  The doc should be updated to discuss in terms of models.